.changeset/sweet-berries-begin.md

```diff
@@ -0,0 +1,38 @@
+---
+'@astrojs/starlight': minor
+---
+
+Adds Expressive Code as Starlight’s default code block renderer
+
+⚠️ **Potentially breaking change:**
+This addition changes how Markdown code blocks are rendered to use [Expressive Code](https://github.com/expressive-code/expressive-code/tree/main/packages/astro-expressive-code).
+If you were already customizing how code blocks are rendered and don't want to use the [features provided by Expressive Code](https://starlight.astro.build/guides/authoring-content/#expressive-code-features), you can preserve the previous behavior by setting the new config option `expressiveCode` to `false`.
+
+If you had previously added Expressive Code manually to your Starlight project, you can now remove the manual set-up in `astro.config.mjs`:
```

Comment:
```suggestion
If you had previously added Expressive Code manually to your Starlight project, you must now remove the manual set-up in `astro.config.mjs`:
```

Reply:
Not necessarily! If someone adds Expressive Code manually, Starlight will skip adding it on its end.

Reply:
Just to provide a bit of context: If you leave your manually added version of `astro-expressive-code` in Astro's `integrations` array, Starlight will detect this and skip adding its own version of the integration. So it's not a "must" per se, but if you want the full Starlight-tailored experience, you should remove your manual one.

Reply:
I think I’ll leave this is as “can” if that’s OK. The Starlight version of EC does differ slightly in its defaults, so some people may wish to keep their current config if they’re happy with it (and there’s no real harm in doing so except potentially some extra dependencies).

Reply:
if both of them (EC and Starlight) use the same config options, we could suggest to merge configs to override starlight defaults and keep the user custom config.

Reply:
> if both of them (EC and Starlight) use the same config options, we could suggest to merge configs to override starlight defaults and keep the user custom config.

I think this is what we are suggesting in the following sentence and showing in the example. By moving your previous manual config into the new `expressiveCode` option, you are overriding the Starlight default options with your changes. This way, you're able to keep your customizations while migrating to the new EC integration built into Starlight.